pkg/analyze/symptoms/rules/storage.go

```diff
@@ -0,0 +1,292 @@
+						return true, nil
+					}
+				}
+				return false, nil
+			}).
+			Relate("scylla-cluster", "scylla-pod", func(c *scyllav1.ScyllaCluster, p *v1.Pod) (bool, error) {
+				return c.Name == p.Labels["scylla/cluster"], nil
+			}).
+			Relate("scylla-pod", "pod-pvc", func(p *v1.Pod, pvc *v1.PersistentVolumeClaim) (bool, error) {
+				for _, volume := range p.Spec.Volumes {
+					if volume.PersistentVolumeClaim != nil {
+						volumePvc := *volume.PersistentVolumeClaim
+						if volumePvc.ClaimName == pvc.Name {
+							return true, nil
+						}
+					}
+				}
+				return false, nil
+			}).
+			Relate("pod-pvc", "storage-class", func(pvc *v1.PersistentVolumeClaim, s *storagev1.StorageClass) (bool, error) {
+				return *pvc.Spec.StorageClassName == s.Name, nil
+			}).
+			Relate("storage-class", "csi-driver", func(s *storagev1.StorageClass, csi *storagev1.CSIDriver) (bool, error) {
+				return s.Provisioner == csi.Name, nil
+			}).
+			Relate("scylla-node", "scylla-cluster", func(n *v1.Node, c *scyllav1.ScyllaCluster) (bool, error) {
+				// TODO: relate the node to the cluster (with node placement rules?)
+				return true, nil
+			}).
+			Relate("node-config", "scylla-node", func(nc *scyllav1alpha1.NodeConfig, n *v1.Node) (bool, error) {
+				return symptoms.MeetsNodeSelectorPlacementRules(n, nc.Spec.Placement.NodeSelector), nil
+			}).
+			// node-config tunes the node on which csi-pod is placed
+			Relate("csi-pod", "node-config", func(csiPod *v1.Pod, nodeConfig *scyllav1alpha1.NodeConfig) (bool, error) {
+				matchesPod := false
+				dirsMounted := false
+				for _, status := range nodeConfig.Status.NodeStatuses {
+					if status.Name == csiPod.Spec.NodeName {
+						matchesPod = true
+					}
+				}
+				if !matchesPod {
+					return false, nil
+				}
+				volumesDir := ""
+				volumesMountPoint := ""
+				for _, container := range csiPod.Spec.Containers {
+					csiContainer := false
+					for _, arg := range container.Args {
+						if strings.Contains(arg, "--volumes-dir=") {
+							volumesDir = strings.Split(arg, "--volumes-dir=")[1]
+							volumesDir = strings.TrimSpace(volumesDir)
+							csiContainer = true
+						}
+					}
+					if csiContainer {
+						for _, mnt := range container.VolumeMounts {
+							if mnt.MountPath == volumesDir {
+								volumesMountPoint = mnt.Name
+							}
+						}
+					}
+				}
+
+				if volumesMountPoint != "" {
+					for _, vol := range csiPod.Spec.Volumes {
+						if vol.Name == volumesMountPoint {
+							for _, mp := range nodeConfig.Spec.LocalDiskSetup.Mounts {
+								if (*vol.HostPath).Path == mp.MountPoint {
+									dirsMounted = true
+								}
+							}
+						}
+					}
+				}
+
+				return !dirsMounted, nil
+			}))
+
+	// Detects the non-existent device just by conditions
+	nodeConfigNonexistentDevice := symptoms.NewSymptom(
+		"NodeConfig non-existent device condition",
+		[]string{"NodeConfig configured with a non-existent device"},
+		[]string{"fix NodeConfig"},
```

Comment:
Let's make this at least a bit more precise. Manifests snippets that I suggested earlier would be very helpful here, but until we have those, let's try to be more descriptive.

Reply:
Done